src/test/java/io/vertx/proton/FutureHandler.java

```diff
@@ -0,0 +1,81 @@
+package io.vertx.proton;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+abstract public class FutureHandler<T, X> implements Future<T>, Handler<X> {
+
+  protected ExecutionException exception;
+  protected T result;
+  protected CountDownLatch latch = new CountDownLatch(1);
+
+  public static <T> FutureHandler<T, T> simple() {
+    return new FutureHandler<T, T>() {
+      @Override
+      synchronized public void handle(T t) {
+        result = t;
+        latch.countDown();
+      }
+    };
+  }
+
+  public static <T> FutureHandler<T, AsyncResult<T>> asyncResult() {
+    return new FutureHandler<T, AsyncResult<T>>() {
+      @Override
+      synchronized public void handle(AsyncResult<T> t) {
+        if (t.succeeded()) {
+          result = t.result();
+        } else {
+          exception = new ExecutionException(t.cause());
+        }
+        latch.countDown();
+      }
+    };
+  }
+
+  @Override
+  abstract public void handle(X t);
+
+  public T get() throws InterruptedException, ExecutionException {
+    latch.await();
+    return result();
+  }
+
+  private T result() throws ExecutionException {
+    synchronized (this) {
+      if (exception != null) {
+        throw exception;
+      }
+      return result;
+    }
+  }
+
+  public T get(long timeout, TimeUnit unit) throws InterruptedException, TimeoutException, ExecutionException {
```

Comment:
Those methods are implementing the java.util.concurrent.Future interface and so can't really be renamed. It is test-only code that the tests use specifically to block, is it an issue to leave it as is?